src/Microsoft.Azure.ServiceBus/Message.cs

```diff
@@ -6,1136 +6,200 @@ namespace Microsoft.Azure.ServiceBus
-        }
-
-        /// <summary>Gets or sets the identifier of the correlation.</summary>
-        /// <value>The identifier of the correlation.</value>
-        /// <exception cref="System.ObjectDisposedException">Thrown if the message is in disposed state.</exception>
-        public string CorrelationId
-        {
-            get
-            {
-                this.ThrowIfDisposed();
-                return this.correlationId;
-            }
-
-            set
-            {
-                this.ThrowIfDisposed();
-                this.correlationId = value;
-                if (value == null)
-                {
-                    this.ClearInitializedMember(MessageMembers.CorrelationId);
-                }
-                else
-                {
-                    this.initializedMembers |= MessageMembers.CorrelationId;
-                }
-            }
-        }
-
-        /// <summary>Gets or sets the identifier of the session.</summary>
-        /// <value>The identifier of the session.</value>
-        /// <exception cref="System.ObjectDisposedException">Thrown if the message is in disposed state.</exception>
-        public string SessionId
-        {
-            get
-            {
-                this.ThrowIfDisposed();
-                return this.sessionId;
-            }
-
-            set
-            {
-                this.ThrowIfDisposed();
-                this.CopySessionId(value);
-                this.PartitionKey = value;
-            }
-        }
-
-        /// <summary>Gets or sets the session identifier to reply to.</summary>
-        /// <value>The session identifier to reply to.</value>
-        /// <exception cref="System.ObjectDisposedException">Thrown if the message is in disposed state.</exception>
-        public string ReplyToSessionId
-        {
-            get
-            {
-                this.ThrowIfDisposed();
-                return this.replyToSessionId;
-            }
-
-            set
-            {
-                this.ThrowIfDisposed();
-                ValidateSessionId(value);
-                this.replyToSessionId = value;
-                if (value == null)
-                {
-                    this.ClearInitializedMember(MessageMembers.ReplyToSessionId);
-                }
-                else
-                {
-                    this.initializedMembers |= MessageMembers.ReplyToSessionId;
-                }
-            }
-        }
-
-        /// <summary>Gets the number of deliveries.</summary>
-        /// <value>The number of deliveries.</value>
-        /// <exception cref="System.ObjectDisposedException">Thrown if the message is in disposed state.</exception>
-        /// <exception cref="System.InvalidOperationException">Thrown if the message has not been delivered by ServiceBus.</exception>
-        public int DeliveryCount
-        {
-            get
-            {
-                this.ThrowIfDisposed();
-                this.ThrowIfNotReceived();
```

Comment:
Should this be removed? i.e. Should the default behavior be to throw when trying to access a property that would only be set when a message is received?